src/exceptions.py

```diff
@@ -4,13 +4,37 @@
 # See LICENSE file for licensing details.
 
 
-class AdminUserCreationError(Exception):
+class MongoError(Exception):
+    """Common parent for Mongo errors, allowing to catch them all at once."""
+
+    pass
+
+
+class AdminUserCreationError(MongoError):
     """Raised when a commands to create an admin user on MongoDB fail."""
 
     pass
 
 
-class ApplicationHostNotFoundError(Exception):
+class ApplicationHostNotFoundError(MongoError):
     """Raised when a queried host is not in the application peers or the current host."""
 
     pass
```

Comment:
NOTE: The changes above do not relate to the PR (and I can roll them back in case it may be an issue).
Since I've implemented secrets-related exceptions accordingly, I altered the existing errors for consistency.
Note that the change is seamless and fully backwards compatible.

***Rationale:***
I find it very useful when group of exceptions can be caught alltogether, when we don't want to handle a specific issue. Typically for usages such as

```
try:
    connect_to_whatever_db()
except WhateverDBError:
    # We don't care for the specifics this time, but only that the error occured on a service level
    logging.error("DB connection failed")
except NetworkError:
    logging.error("A network issue occured")   
```
rather than
```
try:
    connect_to_whatever_db()
except (
        WhateverDBErrorConnectionError,
        WhateverDBErrorAuthenticationError,
        WhateverDBErrorInvalidRoleErrorm,
#        IForgotYetAnotherWhateverDBConnectionRelatedErrrorToCatchSoItBubblesUp
):
    logging.error("Connection failed")
```
(Typically, in our case: is it a MongoDB error? Or is it an error on the level of our charm?)